From 50f1512c22958cafed83aea44ac38492e2dac162 Mon Sep 17 00:00:00 2001
From: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
Date: Fri, 4 Oct 2024 10:27:04 +0000
Subject: [PATCH] Fix BT crash due to BT audio HAL dependency

Bluetooth LE Audio profiles requires Bluetooth Audio AIDL HAL to
be present for successful initialization. This is causing
continuous BT crash.

Add Bluetooth Audio AIDL HAL as part of btusb product.mk for
successful BT LE Audio profile initialization. Now BT works fine.

Tests done:
1. Flash AAOS
2. Check BT on/off success

Tracked-On: OAM-126100
Signed-off-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 groups/bluetooth/btusb/product.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/groups/bluetooth/btusb/product.mk b/groups/bluetooth/btusb/product.mk
index 0ed48a5..7c10f87 100644
--- a/groups/bluetooth/btusb/product.mk
+++ b/groups/bluetooth/btusb/product.mk
@@ -6,7 +6,8 @@ PRODUCT_COPY_FILES += \
 PRODUCT_PROPERTY_OVERRIDES += bluetooth.rfkill=1
 
 PRODUCT_PACKAGES += \
-    android.hardware.bluetooth-service.default.vbt 
+    android.hardware.bluetooth-service.default.vbt \
+    android.hardware.bluetooth.audio-impl
 
 {{#ivi}}
 PRODUCT_PACKAGE_OVERLAYS += $(INTEL_PATH_COMMON)/bluetooth/overlay-car
-- 
2.34.1

